src/core/project.js

```diff
@@ -42,7 +44,7 @@ export function projectToJSON(p: Project): ProjectJSON {
 }
 
 export function projectFromJSON(j: ProjectJSON): Project {
-  return fromCompat(COMPAT_INFO, j);
+  return fromCompat(COMPAT_INFO, j, upgrades);
```

Comment:
Nice! I think this is the first use of the `upgrades` API. @wchargin and I have gone back and forth a bit on the quality of this API, I'd be curious to hear what you thought of this interface.